url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/neworder/addpizza/AddPizzaViewCL.java

```diff
@@ -0,0 +1,256 @@
+package neworder.addpizza;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import model.PizzaFoodItem;
+import model.Topping;
+import neworder.NewOrderView;
+import neworder.addpizza.AddPizzaView.AddPizzaInChan;
+import neworder.addpizza.AddPizzaView.AddPizzaOutChan;
+import viewcontroller.GeneralView;
+
+public class AddPizzaViewCL extends AddPizzaView {
+
+	/**
+	 * This view's scanner.
+	 */
+	private Scanner scan;
+	
+	
+	/**
+	 * The default constructor.
+	 */
+	public AddPizzaViewCL() {
+		
+		scan = new Scanner( System.in );
+		
+	}
+	
+
+
+	/**
+	 * @see viewcontroller.GeneralView#displayString(java.lang.String, viewcontroller.GeneralView.OutputChannel)
+	 */
+	public void displayString(String message, OutputChannel outChannel) {
+		
+		// Pretty much everything is just written to the terminal, but
+		// we'll insert some tags to help with tracing.
+		switch( (AddPizzaOutChan)outChannel ) {
+		
+		case OCInstructions:
+			System.out.println( "Instructions: " + message );
+			break;
+			
+		case OCError:
+			System.err.println( "Error: " + message );
+			break;
+		
+		}
+		
+		System.out.println();
+		
+	}
+
+
+	/**
+	 * @see viewcontroller.GeneralView#displayObject(java.lang.Object, viewcontroller.GeneralView.OutputChannel)
+	 */
+	public void displayObject(Object object, OutputChannel outChannel) {
+		
+		if( outChannel == AddPizzaOutChan.OCDisplayPizza) {
+			
+			// The object is a pizza.
+			PizzaFoodItem pizza = (PizzaFoodItem)object;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;object&#39; (index: 0 | type: Object). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;outChannel&#39;. The same parameter type is validated here for example: url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/customereditor/CustomerEditorViewGUI.java:400. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).